tests/unit/test_index.py

```diff
@@ -280,3 +280,24 @@ def test_request_retries(caplog):
         'Could not fetch URL http://localhost: Retry error - skipping'
         in caplog.text
     )
+
+
+_path = "/path with space/and\"[more]:%3A?request=test"
+_clean_path = "/path%20with%20space/and%22%5Bmore%5D%3A%3A?request=test"
```

Comment:
You asked about `path` and `_clean_path`. To keep the test simpler, I would just use strings in your test cases rather than variables. Also, instead of using a string with many different problem characters, I would make it so that each test case tests just test one aspect or type of character. For example:

* `https://example.com/space space`
* `https://example.com/already%3Aquoted`

When you have lots of problem characters at once, it's hard to tell which aspect is failing when the test fails (or whether it's multiple aspects).

Also, if you wanted, you could put a comment before each test case.

Reply:
Done.